File: algorithms/Java/src/datastructure/linked_list/283.java
package datastructure.linked_list;

/**
 * @author: batteria
 * @version: 1.0
 * @since: 2020/12/12
 * @description: 26.remove-duplicates-from-sorted-array
 */
class Solution {
    public int removeDuplicates(int[] nums) {
        int size = nums.length;
        if (size == 0) return 0;
        int last = nums[0], left = 1, result = 1;
        for (int i = 1; i < size; i++) {
            int temp = nums[i];
            if (temp != last) {
                last = temp;
                nums[left++] = temp;
                result++;
            }
        }
        return result;
    }

    public static void main(String[] args) {

    }
}

File: algorithms/Kotlin/src/datastructure/linked_list/283.kt
package datastructure.linked_list

/**
 * @author: batteria
 * @version: 1.0
 * @since: 2020/12/12
 * @description: 26.remove-duplicates-from-sorted-array
 */
fun removeDuplicates(nums: IntArray): Int {
    if (nums.isEmpty()) return 0
    var result = 1
    var left = 1
    var last = nums[0]
    for (i in nums.indices) {
        if (nums[i] != last) {
            last = nums[i]
            nums[left++] = nums[i]
            result++
        }
    }
    return result
}

fun main() {

}